build_config/webpack.dev.conf.js

```diff
@@ -11,6 +11,11 @@ module.exports = {
         exclude: /node_modules/,
         loader: 'babel-loader?cacheDirectory=true'
       },
+      {
```

Comment:
Can I recommend introducing typechecking to our watch and build tasks? There are a couple of ways I know of –

- use [`fork-ts-checker-webpack-plugin`](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin)
- use the approach detailed by @twrichards in [this thread](https://chat.google.com/room/AAAAWaoV0IE/uozJCuL5Ej4), and use either `&` or `run-p` to run `tsc` alongside the watch process, and on build (we'll need to change the build script, and fail fast when the build fails, if that isn't already the case.)

In this way we've got the best possible view of our compiler's opinion during `watch`, and we can't ship type errors in deployables.

Reply:
(More detail: `?transpileOnly=true'` ensures that ts-loader will simply strip the type information from .ts files as it converts them to js. This is great for fast builds and a good developer experience, but explicitly skipping the typecheck leaves us vulnerable to errors.)